pkg/cache/memcached.go

```diff
@@ -143,9 +123,43 @@ func (c *Memcached) fetch(ctx context.Context, keys []string) (found []string, b
 	return
 }
 
+// FetchKey gets a single key from the cache
+func (c *Memcached) FetchKey(ctx context.Context, key string) ([]byte, bool) {
+	select {
+	case <-ctx.Done():
+		return nil, false
+	default:
+	}
+
+	const method = "Memcache.Get"
+	var item *memcache.Item
+	err := measureRequest(ctx, method, c.requestDuration, memcacheStatusCode, func(_ context.Context) error {
+		var err error
+		item, err = c.memcache.Get(key, memcache.WithAllocator(cacheBufAllocator))
+		if err != nil {
+			if errors.Is(err, memcache.ErrCacheMiss) {
+				level.Debug(c.logger).Log("msg", "Failed to get key from memcached", "err", err, "key", key)
+			} else {
+				level.Error(c.logger).Log("msg", "Error getting key from memcached", "err", err, "key", key)
+			}
+		}
+		return err
+	})
+	if err != nil {
```

Comment:
Not familiar with the nuances of the error, but should we also check the context before returning, IE if it has been cancelled simulate a nil/false response?

Reply:
in the case of the panic in question this won't matter. i copied this pattern of "check once at the beginning" from dskits memcached client.

Not sure if checking with more frequency is worth it.